### ISSUE TITLE
[NuGet] VSM restores packages when restore is disabled

### ISSUE THREAD

**Issue by garuma** [mono/monodevelop#6588]
When the options "Automatically restore packages when opening a solution" and "Check for package updates when opening solution" are unchecked in the IDE, I can still see VSM trying to restore packages when an external changes to a .sln or .csproj cause the solution to be reloaded.

**Comment by mrward** [comment]
When the solution file is edited externally or in the text editor then the IDE will load the solution again so a restore will happen. So this is currently by design. Switching branches may cause different NuGet packages to be used so a restore may need to happen.

**Comment by garuma** [comment]
Yep that make sense but I would assume that by having those options disabled the "opening solution" would imply "reload solution" as well.

The usecase for me is that I typically switch branch and then build on the command line (with restore) before I go back to the IDE so that's why I would like to disable the extra step in that case too.

**Comment by mrward** [comment]
Sorry I misread - I thought the options were checked not unchecked.

Are you using sdk style projects? A restore currently happens when a sdk project is reloaded since this handles the case where the project file is edited in the text editor.

**Comment by garuma** [comment]
Yep we are SDK-style (completely) now. I guess it can't be currently differentiated where the edit came from?

**Comment by mrward** [comment]
Would need to check if we can distinguish the two. Disabling automatic restore on opening usually means you do not want any automatic restore so we could just check the setting and not restore here if a project file is edited or reloaded. Using a packages.config file and unchecking the automatic restore on open would disable all automatic restores. When I added support for restoring on editing the project file I was in two minds about whether the automatic restore on open would affect the behaviour.
